pkg/credentials/iam_aws.go

```diff
@@ -104,7 +104,11 @@ func (m *IAM) Retrieve() (Value, error) {
 			},
 		}
 
-		return creds.Retrieve()
+		stsWebIdentityCreds, err := creds.Retrieve()
+		if err == nil {
+			m.SetExpiration(creds.Expiration(), 0)
```

Comment:
```suggestion
			m.SetExpiration(creds.Expiration(), DefaultExpiryWindow)
```

Reply:
DefaultExpiryWindow has already been applied.

`creds.Expiration()` returns `m.expiration` which is set by `m.SetExpiration(a.Result.Credentials.Expiration, DefaultExpiryWindow)` in `STSWebIdentity`

I figured we didn't want to add DefaultExpiryWindow twice.

Reply:
If this is a new value, we should apply here as well.